### PR TITLE
MMA-9932 added default configs resource 

### PR DESCRIPTION
**Pull request by thomaskwscott**
New resource at:

/v3/clusters/{clusterId}/topics/{topicName}/default-configs

that fetches default configs that would be applied should topicName be created.

Note: this applies to new topics only, requests where topicName is pre-existing fail with 400 error.

This is covered with unit and integration tests

### CODE REVIEW

**Comment by thomaskwscott** [comment]
@srpanwar-confluent @dimitarndimitrov bump for review.

File: kafka-rest/src/main/java/io/confluent/kafkarest/controllers/DefaultTopicConfigManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        .thenApply(cluster -> checkEntityExists(cluster, "Cluster %s cannot be found.", clusterId))
+        .thenCompose(
+            cluster ->
+                KafkaFutures.toCompletableFuture(
+                    adminClient.createTopics(
+                        singletonList(new NewTopic(topicName, Optional.empty(), Optional.empty())),
+                        new CreateTopicsOptions().validateOnly(true)
+                    ).config(topicName)))
+        .handle((result, ex) -> {
+              // InvalidRequestException can happen in the case of multi-tenancy, where passing
```
Comment (srpanwar-confluent) [inline kafka-rest/src/main/java/io/confluent/kafkarest/controllers/DefaultTopicConfigManagerImpl.java:76]: `multi-tenancy` is not available in AK. Maybe we should remove its reference from the comment.

**Comment by srpanwar-confluent** [comment]
PR looks good to me!. Few things 
1. Should we move this `ce-kafka-rest-extension`? There is nothing proprietary with the API per se, so adding it here should be fine.
2. Reason I am thinking of 1) is that there is an edge case in ccloud, where, if the cluster is close to hitting the partition limit this call will fail. This is because #defaultpartition is > # partition remaining. This is not fixed in the original implementation as well, so I don't we should try to fix it in this PR.  It may require a bit more work to pipe the remaining partition cloud. Something to keep a note of though.

**Review by srpanwar-confluent — APPROVED** [review]

**Comment by thomaskwscott** [comment]
closed in favour of https://github.com/confluentinc/ce-kafka-rest/pull/149

**Pull request closed without merging**